libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/route/NavigationMapRoute.java

```diff
@@ -326,6 +336,19 @@ public void showAlternativeRoutes(boolean alternativesVisible) {
     toggleAlternativeVisibility(alternativesVisible);
   }
 
+  public void addProgressChangeListener(MapboxNavigation navigation) {
+    this.navigation = navigation;
+    navigation.addProgressChangeListener(progressChangeListener);
+  }
+
+  //TODO Is this necessary? It's never used
+  public void removeProgressChangeListener(MapboxNavigation navigation) {
```

Comment:
If we choose to keep add / remove progress change listener `public`, I'd say this is necessary.  We need to keep in mind that public APIs may not be used by us, but could be necessary for developers using these classes as a standalone UI component.